src/types/connector.ts

```diff
@@ -1,11 +1,23 @@
-import { InstantSearch } from './instantsearch';
+import { Hits, InstantSearch } from './instantsearch';
+import { InsightsClient } from './insights';
 
-export type RenderOptions<T = unknown> = {
-  widgetParams: T;
+// export type RenderOptionsWithInsights<T> = RenderOptions<T> implements
```

Comment:
No sure whether I should do something like this instead of `insights?`

Reply:
`RenderOptions` is the most generic type that other types have to extend. We should add on it only the common options.